### PR TITLE
small fixes

### PR DESCRIPTION
**Pull request by cmkohnen**
Small fixes across the codebase from my testing and pylint analysis

### CODE REVIEW

File: src/file_io.py [hunk 1]
```diff
@@ -18,18 +18,18 @@
 import numpy
 
 
-def save_item(file, item):
+def save_item(file, item_):
```
Comment (Sjoerd1993) [inline src/file_io.py:21]: Just as a general point, the trailing underscore is to distinguish it from item methods in e.g. dictionaries?

File: src/operations.py [hunk 2]
```diff
@@ -189,9 +189,9 @@ def shift_vertically(item, xdata, ydata, left_scale, right_scale, items):
         ymin = min(x for x in previous_ydata if x != 0)
         ymax = max(x for x in previous_ydata if x != 0)
         if item.yposition == "left":
-            linear = (left_scale == "linear")
+            linear = left_scale == "linear"
```
Comment (Sjoerd1993) [inline src/operations.py:192]: Any specific reason for getting rid of the brackets? Personally I feel the brackets make it a bit more readable, as it distinguishes the comparison  (==) from the assignment (=) a bit visually.

File: src/utilities.py [hunk 3]
```diff
@@ -214,3 +206,4 @@ def get_next_color(items):
     for color in color_cycle:
         if color not in used_colors:
             return color
+    return "000000"
```
Comment (Sjoerd1993) [inline src/utilities.py:209]: In principle it should never reach this last return value. If all colors in the cycle are full, it should have emptied the color_cycle again.

But I don't mind this return value as a potential safeguard anyway in unforeseen edge cases.

File: src/operations.py [hunk 4]
```diff
@@ -189,9 +189,9 @@ def shift_vertically(item, xdata, ydata, left_scale, right_scale, items):
...
         if item.yposition == "right":
```
Comment (Sjoerd1993) [inline src/operations.py:193]: These are saved as enums now, with integer values:

```
        if item.yposition == 0:
            linear = left_scale == 0
        if item.yposition == 1:
            linear = right_scale == 0
```

File: src/operations.py [hunk 5]
```diff
@@ -189,9 +189,9 @@ def shift_vertically(item, xdata, ydata, left_scale, right_scale, items):
         ymin = min(x for x in previous_ydata if x != 0)
```
Comment (Sjoerd1993) [inline src/operations.py:189]: If item_ with the trailing underscore is used in the other methods, maybe it makes sense to do this here as well. Unless there's a reason for changing to that elsewhere that I missed.

**Review by Sjoerd1993 — CHANGES_REQUESTED** [review]

**Comment by Sjoerd1993** [comment]
There's also something out of wack with the plots when doing operations on multiple data sets. Basically it only plots the latest data set. Seems to be in the latest main commit as well, so probably a regression from last refactor. 
Will probably take a look into that tomorrow.

[Uploading Screencast from 2023-08-24 20-59-23.webm…]()

File: src/file_io.py (hunk 1 above)
Comment (cmkohnen) [reply src/file_io.py:21]: A trailing underscore is a python convention to distinguish between definitions in a higher scope.

Here, item is already defined, hence appending it.

**Comment by cmkohnen** [comment]
> There's also something out of wack with the plots when doing operations on multiple data sets. Basically it only plots the latest data set. Seems to be in the latest main commit as well, so probably a regression from last refactor. Will probably take a look into that tomorrow.
> 
> Uploading Screencast from 2023-08-24 20-59-23.webm…

Fixed the issue, only the last item emitted a notify signal for x and y data

**Review by Sjoerd1993 — APPROVED** [review]

**Pull request merged**